src/Core/Filter/FrontEndObject/CustomerFilter.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Filter\FrontEndObject;
+
+use PrestaShop\PrestaShop\Core\Filter\HashMapWhitelistFilter;
+
+/**
+ * Filters Customer objects that will be sent to the client
+ */
+class CustomerFilter extends HashMapWhitelistFilter
+{
+    public function __construct()
+    {
+        $whitelist = array(
+            "addresses",
+            "ape",
+            "birthday",
+            "company",
+            "email",
+            "firstname",
+            "gender" => (new HashMapWhitelistFilter())
+                ->whitelist(array(
+                    "type",
+                    "name",
+                )),
+            "is_logged",
+            "lastname",
+            "newsletter",
+            "newsletter_date_add",
+            "optin",
+            "siret",
+            "website",
+        );
+
```

Comment:
@mickaelandrieu 

what about:
`$whitelist = \Hook::exec('actionCustomerFieldsWhiteListModifier', ['whitelist' => &$whitelist]);`

?

Reply:
well, I prefer we include the hook dispatcher as dependency and do:

```php
$whitelist = $this->hookDispatcher->dispatch('actionXxx', ['whitelist' => ... ]);
```

no more global static used, and easier to unit test :+1:

Reply:
@mickaelandrieu yeah sure, i forgot about `hookDispatcher`, you get the idea tho :)